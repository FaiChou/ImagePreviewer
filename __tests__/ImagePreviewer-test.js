import React from 'React';
import ReactTestRenderer from 'react-test-renderer';
import ImagePreviewer from '../index';
const ImgWidth = 300;
const ImgHeight = ImgWidth * 0.6;

describe('ImagePreviewer', () => {
  it('renders correctly', () => {
    const instance = ReactTestRenderer.create(
      <ImagePreviewer
        source={{uri: 'http://o7bkcj7d7.bkt.clouddn.com/2.png'}}
        style={{
          width: ImgWidth,
          height: ImgHeight,
        }}
        resizeMode="stretch"
      />
    );

    expect(instance.toJSON()).toMatchSnapshot();
  });
});