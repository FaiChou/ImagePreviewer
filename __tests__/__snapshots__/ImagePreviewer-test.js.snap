// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePreviewer renders correctly 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 180,
        "width": 300,
      }
    }
    testID={undefined}
  >
    <Image
      resizeMode="stretch"
      source={
        Object {
          "uri": "http://o7bkcj7d7.bkt.clouddn.com/2.png",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 180,
          "width": 300,
        }
      }
    />
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "http://o7bkcj7d7.bkt.clouddn.com/2.png",
          }
        }
        style={
          Object {
            "flex": 1,
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": "100%",
          }
        }
      />
    </View>
  </Modal>
</View>
`;
